### PR TITLE
generate source links in the documentation

### PR DESCRIPTION
**Pull request by yole**

### CODE REVIEW

File: src/main.kt [hunk 1]
```diff
@@ -14,6 +13,10 @@ class DokkaArguments {
     ValueDescription("<path>")
     public var src: String = ""
 
+    Argument(value = "srcLink", description = "Mapping between a source directory and a Web site for browsing the code")
+    ValueDescription("<path>=<url>[#lineSuffix]")
+    public var srcLink: String = ""
+
```
Comment (orangy) [inline src/main.kt]: Can't there be multiple mappings? E.g. when project consists of multiple source repositories checked out into single folder structure.

File: src/main.kt [hunk 2]
```diff
@@ -40,6 +43,15 @@ class DokkaArguments {
 
 }
 
+class SourceLinkDefinition(val path: String, val url: String, val lineSuffix: String?)
+
+private fun parseSourceLinkDefinition(srcLink: String): SourceLinkDefinition {
+    val urlAndLine = srcLink.substringAfter("=")
```
Comment (orangy) [inline src/main.kt]: this should work here:
val (path, urlAndLine) = srcLink.split("=")

File: src/main.kt (hunk 1 above)
Comment (yole) [inline src/main.kt]: The underlying logic supports multiple mappings, but I'm not such a big fan of trying to represent this in a command line parameter.

Comment (orangy) [inline src/main.kt]: Just use ":" between mappings :) Like in classpath.

File: src/main.kt (hunk 2 above)
Comment (yole) [inline src/main.kt]: Done

File: src/main.kt (hunk 1 above)
Comment (yole) [inline src/main.kt]: : is valid in URLs

**Pull request merged**